firmware/mcal/cli/periph/pwm.h

```diff
@@ -30,15 +30,30 @@ class PWMOutput : public shared::periph::PWMOutput {
         std::cout << "Setting PWM " << name_ << " to " << duty_cycle_ << "%"
                   << std::endl;
     }
+
     float GetDutyCycle() override {
         std::cout << "PWM " << name_ << " has duty cycle " << duty_cycle_ << "%"
                   << std::endl;
         return duty_cycle_;
     }
 
+    void SetFrequency(float frequency) override {
+        frequency_ = frequency;
```

Comment:
can you clamp the frequency to ensure its positive? You can use `Clamper` to do one sided clamping; see the comment in `shared/util/mappers/Clamper`

Reply:
Actually, it makes more sense just to use `freq = max(0, freq)`